src/combat.mjs

```diff
@@ -1,28 +1,52 @@
 import {item_roll, Logger} from "./util.mjs";
 import {socket} from "./module.mjs";
-import {COMBAT_HEALTH_ESTIMATE, ID_MAP, MODULE_ID} from "./constants.mjs";
+import {COMBAT_ENABLED, COMBAT_HOOKS, COMBAT_HEALTH_ESTIMATE, ID_MAP, MODULE_ID} from "./constants.mjs";
 import { actor_to_discord_ids } from "./sync.mjs";
 
+const onCombatStart = async (combat, updateData) =>  {
+    const roundRender = parseCombatRound({ ...combat, ...updateData })
+    const turnRender = parseTurn(combat, updateData) 
+    socket.emit('combat', roundRender+turnRender)
+}
+const onCombatTurn = async (combat, updateData, updateOptions) => {
+    if (updateOptions.direction < 1) return
+    const turnRender = parseTurn(combat, updateData) 
+    socket.emit('combat', turnRender)
+}
+const onCombatRound = async (combat, updateData, updateOptions) => {
+    if (updateOptions.direction < 1) return
+    const roundRender = parseCombatRound({ ...combat, ...updateData }, updateOptions)
+    const turnRender = parseTurn(combat, updateData) 
+    socket.emit('combat', roundRender+turnRender)
+}
 
 export function set_combat_hooks() {
     Logger.info("Setting Combat Hooks.")
 
-    Hooks.on("combatStart", async (combat, updateData) => {
-        const roundRender = parseCombatRound({ ...combat, ...updateData })
-        const turnRender = parseTurn(combat, updateData) 
-        socket.emit('combat', roundRender+turnRender)
-    })
-    Hooks.on("combatTurn", async (combat, updateData, updateOptions) => {
-        if (updateOptions.direction < 1) return
-        const turnRender = parseTurn(combat, updateData) 
-        socket.emit('combat', turnRender)
-    })
-    Hooks.on("combatRound", async (combat, updateData, updateOptions) => {
-        if (updateOptions.direction < 1) return
-        const roundRender = parseCombatRound({ ...combat, ...updateData }, updateOptions)
-        const turnRender = parseTurn(combat, updateData) 
-        socket.emit('combat', roundRender+turnRender)
-    })
+    let combatHooks = game.settings.get(MODULE_ID, COMBAT_HOOKS)
```

Comment:
I don't believe it needs to be stored as a setting. If you just store them as a variable scoped to the top level of module.mjs I believe it will persist for as long as the hook does.

Reply:
updated, seems to work. They go to default when you open the world but that's fine since they are registered if enabled at that point and then overridden.